### PR TITLE
Express notes api

### PR DESCRIPTION
**Pull request by AndyPark20**
--------------------------------------GET----------------------------------------
challenge get
-----------------------------------GETbyID------------------------------------
challenge getbyId
-----------------------------------POST------------------------------------
challenge post
-----------------------------------DELETE------------------------------------
challenge delete
-----------------------------------PUT------------------------------------
challenge put

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  }
+})
+
+//Put request and replace note by ID
+app.put('/api/notes/:id', (req, res) => {
+  const userNum = parseInt(req.params.id, 10);
+
+  if (userNum !== Math.abs(userNum)){
+    return(res.status(400).json(`Error: Sorry, but the ID needs to be a positive integer`))
+  } else if (Object.entries(req.body).length === 0) {
```
Comment (scbowler) [inline express-notes-api/index.js:88]: Like above you should just be checking directly for the `content` property on `req.body`

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    return(res.status(400).json(`Error: Please fill out the required field`))
+  }
+
+  if (!data.notes[userNum]) {
+    res.status(404).json(`Error: Sorry, no matching ID# ${userNum} found!`)
+  } else if(data.notes[userNum]) {
```
Comment (scbowler) [inline express-notes-api/index.js:94]: Just an `else` here, you don't need to check the opposite of what was just checked

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, but some changes are needed

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
+const express = require('express');
+const fs = require('fs');
+const { get } = require('http');
+const { send } = require('process');
```
[the remark this answers is not in the document]
Comment (AndyPark20) [reply express-notes-api/index.js:4]: That's really weird how it got there....but definitely I will go ahead and delete these!

File: express-notes-api/index.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+
+//get request to pull specific notes based on Id number
+app.get('/api/notes/:id', (req, res, next) => {
+  const dataNotes = data.notes;
+  const inputNumber = parseInt(req.params.id, 10);
+
+  if (inputNumber !== Math.abs(inputNumber)) {
+    res.status(400).json(`Error: Sorry, but the ID needs to be a positive integer`)
+  } else if (inputNumber === Math.abs(inputNumber)) {
```
[the remark this answers is not in the document]
Comment (AndyPark20) [reply express-notes-api/index.js:30]: Noted!

File: express-notes-api/index.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    } else {
+      return(res.status(404).json(`Error: Sorry, no matching ID# ${inputNumber} found!`))
+    }
+  }
+})
+
+// post request to create a new note
+app.post('/api/notes', (req, res, next) => {
+  let nextIds = data.nextId++;
+  data.notes[nextIds] = req.body;
```
[the remark this answers is not in the document]
Comment (AndyPark20) [reply express-notes-api/index.js:42]: Well noted on this one as well. I will go ahead and place these under the if statement checking for the body content.

File: express-notes-api/index.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+  if (Object.entries(req.body).length === 0) {
+    res.status(400).json(`Error: Please fill out the required field!`)
+  } else if (req.body !== {}) {
```
[the remark this answers is not in the document]
Comment (AndyPark20) [reply express-notes-api/index.js:46]: Well understood!

File: express-notes-api/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (scbowler) [reply express-notes-api/index.js]: VS Code will auto complete stuff and add those, it is annoying!! lol

File: express-notes-api/index.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+      return(res.status(404).json(`Error: Sorry, no matching ID# ${inputNumber} found!`))
+    }
+  }
+})
+
+// post request to create a new note
+app.post('/api/notes', (req, res, next) => {
+  if (!req.body.content) {
+    return(res.status(400).json(`Error: Please fill out the required field!`))
+  } else if (req.body.content) {
```
Comment (scbowler) [inline express-notes-api/index.js:41]: Same same, you don't need to check the opposite of what was just checked. Just an `else` here

**Review by scbowler — CHANGES_REQUESTED** [review]
So close! One more little fix

**Review by scbowler — APPROVED** [review]
Great job!!! 👍

**Pull request merged**